client/platform/web-girder/router.ts

```diff
@@ -51,10 +52,21 @@ const router = new Router({
           beforeEnter,
         },
         {
-          path: ':_modelType?/:_id?',
-          name: 'home',
+          path: '',
           component: Home,
-          beforeEnter,
+          children: [
+            {
+              path: 'shared',
+              name: 'shared',
+              component: DataShared,
```

Comment:
Would  it make sense to do a global beforeEach here?

Reply:
I don't think it makes a difference. You can either put a beforeEnter in 2 places or you can check inside of a global beforeEach that the next route isn't the login route to prevent an infinite redirect.

Reply:
OK that makes sense. I don't personally have a preference so I will add the `beforeEnter`. It does look as though it needs to be added to each individual route, not the parent.